lib/index.js

```diff
@@ -356,7 +342,7 @@ ABI.rawDecode = function (types, data) {
       }
 
       cur = data.slice(dataOffset)
-    } else if (type.match(/\w+\[\d+\]/)) {
+    } else if (type.match(/\w+\[\d+\]$/)) {
```

Comment:
I think this should be just `/\[\d+\]$/` - otherwise it does not match `uint[4][3]`. Does this library have tests for such things?